diff --git a/Makefile b/Makefile
index 481e8b9..b7ecda2 100644
--- a/Makefile
+++ b/Makefile
@@ -177,7 +177,7 @@ else ifneq (,$(findstring rpi,$(platform)))
       endif
    else ifneq (,$(findstring rpi5,$(platform)))
       ifneq (,$(findstring rpi5_64,$(platform)))
-         CPUFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76
+         CPUFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76 -DPAGESIZE=16384
       else
          CPUFLAGS += -march=armv8-a+crc+crypto -mtune=cortex-a76
          ARM_CPUFLAGS = -mfpu=neon-fp-armv8
diff --git a/mupen64plus-rsp-paraLLEl/jit_allocator.cpp b/mupen64plus-rsp-paraLLEl/jit_allocator.cpp
index 4207fab..77412d7 100644
--- a/mupen64plus-rsp-paraLLEl/jit_allocator.cpp
+++ b/mupen64plus-rsp-paraLLEl/jit_allocator.cpp
@@ -36,7 +36,9 @@ Allocator::~Allocator()
 
 static size_t align_page(size_t offset)
 {
-#if defined(__APPLE__) && defined(__aarch64__)
+#if defined(PAGESIZE)
+	size_t pagesize = PAGESIZE - 1;
+#elif defined(__APPLE__) && defined(__aarch64__)
 	size_t pagesize = sysconf(_SC_PAGESIZE) - 1;
 #else
 	size_t pagesize = 4095;
